src/hotspot/cpu/aarch64/register_aarch64.hpp

```diff
@@ -144,21 +146,25 @@ class FloatRegisterImpl: public AbstractRegisterImpl {
   };
 
   // construction
-  inline friend FloatRegister as_FloatRegister(int encoding);
+  inline friend constexpr FloatRegister as_FloatRegister(int encoding);
 
-  VMReg as_VMReg();
+  VMReg as_VMReg() const;
 
   // derived registers, offsets, and addresses
-  FloatRegister successor() const                          { return as_FloatRegister((encoding() + 1) % 32); }
+  FloatRegister successor() const {
+    return as_FloatRegister((encoding() + 1) % (unsigned)number_of_registers);
+  }
```

Comment:
Different from the other two, why? If we need validity checks here, should we not do them with the other types too?

Reply:
The modulo-32 behaviour in `successor()` is because of an ugly hack elsewhere. The aim of this patch is to remove the UB in class Assembler, not to fix anything else. That's just to make this patch as simple as it can be.